server/src/Irony.cpp

```diff
@@ -333,4 +334,34 @@ void Irony::complete(const std::string &file,
     clang_disposeCodeCompleteResults(completions);
     std::cout << ")\n";
   }
+
+}
+
+// Get compile options from JSON database
+void Irony::getCompileOptions(const std::string &projectRoot,
+                              const std::string &file) {
+  std::vector<std::vector<std::string>> flags = getFlags(projectRoot, file);
+
+  if (!flags.empty()) {
```

Comment:
I believe you can spare you the trouble of checking `flags.empty()`

Elisp consider `nil` and `()` the same.

So you can just do:

```cpp
std::cout << "(";

for (compileCommand : flags) {
  std::cout << compileCommand << "\n";
}

std::cout << ")\n";
```